### PR TITLE
feat: Attributes forwarding to message enums and fields

### PR DESCRIPTION
**Pull request by kulikthebird**

### CODE REVIEW

File: sylvia-derive/src/parser/attributes/mod.rs [hunk 1]
```diff
@@ -68,6 +76,21 @@ impl ParsedSylviaAttributes {
...
+
+        if let Some(attr) = result.variant_attrs_forward.first() {
+            if let Some(MsgAttr::Instantiate) = result.msg_attr {
+                emit_error!(
+                    attr.span, "The attribute `sv::attr` is not supported for `instantiate`";
+                    note = "Message `instantiate` is a structure, use `#[sv::msg_attr] instead`";
+                );
+            } else if let Some(MsgAttr::Migrate) = result.msg_attr {
+                emit_error!(
+                    attr.span, "The attribute `sv::attr` is not supported for `migate`";
```
Comment (jawoznia) [inline sylvia-derive/src/parser/attributes/mod.rs:88]:
```suggestion
                    attr.span, "The attribute `sv::attr` is not supported for `migrate`";
```

File: sylvia-derive/src/parser/attributes/attr.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        let msg_type = match msg_type.to_string().as_str() {
+            "exec" => MsgType::Exec,
+            "query" => MsgType::Query,
+            "instantiate" => MsgType::Instantiate,
+            "migrate" => MsgType::Migrate,
+            "reply" => MsgType::Reply,
+            "sudo" => MsgType::Sudo,
+            _ => return Err(Error::new(
+                input.span(),
+                "Invalid message type, expected one of: `exec`, `query`, `instantiate`, `migrate`, `reply` or `sudo`.",
```
Comment (jawoznia) [inline sylvia-derive/src/parser/attributes/attr.rs:81]: Consider adding test cases with `trybuild` for newly introduced errors

**Review by jawoznia — COMMENTED** [review]
Well done. Consider adding test cases with `trybuild` to cover the errors.

Also, please update the docs on `contract` and `interface` macros.

File: sylvia-derive/src/parser/attributes/attr.rs (hunk 2 above)
Comment (kulikthebird) [reply sylvia-derive/src/parser/attributes/attr.rs:63]: Done

File: sylvia-derive/src/parser/attributes/mod.rs (hunk 1 above)
Comment (kulikthebird) [reply sylvia-derive/src/parser/attributes/mod.rs]: Done

**Review by jawoznia — APPROVED** [review]
Please document this attribute in doc comments of `contract` macro.
You can do it on separate PR.

LGTM

**Comment by kulikthebird** [comment]
> Please document this attribute in doc comments of `contract` macro. You can do it on separate PR.
> 
> LGTM

I've just created a new issue: https://github.com/CosmWasm/sylvia/issues/390

**Pull request merged**